sympy/solvers/ode/single.py

```diff
@@ -247,3 +280,345 @@ def inverse(self):
+        >>> pprint(genform)
+            d
+        f(x)*--(l(y)) + g(x) + k(x)*l(y) = 0
+            dy
+        >>> pprint(dsolve(genform, hint = 'almost_linear'))
+                /     //       y*k(x)                \\
+                |     ||       ------                ||
+                |     ||        f(x)                 ||  -y*k(x)
+                |     ||-g(x)*e                      ||  --------
+                |     ||--------------  for k(x) != 0||    f(x)
+        l(y) = |C1 + |<     k(x)                    ||*e
+                |     ||                             ||
+                |     ||   -y*g(x)                   ||
+                |     ||   --------       otherwise  ||
+                |     ||     f(x)                    ||
+                \     \\                             //
+
+
+    See Also
+    ========
+    :meth:`sympy.solvers.ode.single.FirstLinear`
+
+    Examples
+    ========
+
+    >>> from sympy import Function, Derivative, pprint
+    >>> from sympy.solvers.ode import dsolve, classify_ode
+    >>> from sympy.abc import x
+    >>> f = Function('f')
+    >>> d = f(x).diff(x)
+    >>> eq = x*d + x*f(x) + 1
+    >>> dsolve(eq, f(x), hint='almost_linear')
+    Eq(f(x), (C1 - Ei(x))*exp(-x))
+    >>> pprint(dsolve(eq, f(x), hint='almost_linear'))
+                        -x
+    f(x) = (C1 - Ei(x))*e
+
+    References
+    ==========
+
+    - Joel Moses, "Symbolic Integration - The Stormy Decade", Communications
+    of the ACM, Volume 14, Number 8, August 1971, pp. 558
+    """
+    hint = "almost_linear"
+    has_integral = True
+
+    def _matches(self):
+        ## Almost-linear equation of the form f(x)*g(y)*y' + k(x)*l(y) + m(x) = 0
+        eq = expand(self.ode_problem.eq)
+        f = self.ode_problem.func.func
+        x = self.ode_problem.sym
+        order = ode_order(eq, f(x))
+
+        if order != 1:
+            return False
+
+        df = f(x).diff(x)
+        c = Wild('c', exclude=[f(x)])
+        d = Wild('d', exclude=[df, f(x).diff(x, 2)])
+        e = Wild('e', exclude=[df])
+
+        r = collect(eq, [df, f(x)]).match(e*df + d)
+        if r:
+            r2 = r.copy()
+            r2[c] = S.Zero
+            if r2[d].is_Add:
+                # Separate the terms having f(x) to r[d] and
+                # remaining to r[c]
+                no_f, r2[d] = r2[d].as_independent(f(x))
+                r2[c] += no_f
+            factor = simplify(r2[d].diff(f(x))/r[e])
+            if factor and not factor.has(f(x)):
+                r2[d] = factor_terms(r2[d])
+                u = r2[d].as_independent(f(x), as_Add=False)[1]
+                r2.update({'a': e, 'b': d, 'c': c, 'u': u})
+                r2[d] /= u
+                r2[e] /= u.diff(f(x))
+                self.coeffs = r2
+                return True
+
+        return False
+
+    def _get_general_solution(self, *, simplify: bool = True):
+        return super()._get_general_solution()
```

Comment:
There is no need to override a method in a subclass of it is just going to defer to super

Reply:
Thank you for letting me know.